extensions/amp-analytics/0.1/events.js

```diff
@@ -414,6 +421,29 @@ export class VisibilityTracker extends EventTracker {
   }
 
   /**
+   * @return {!Promise}
+   * @visibleForTesting
+   */
+  createReportReadyPromise_() {
+    const viewer = this.root.getViewer();
+    let resolver = null;
```

Comment:
! it looks much better!
other than using extra memory, an extra line of code. Is there anything I am missing when you says leak?